core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -64,4 +77,21 @@ public static boolean hasDeletes(FileScanTask task) {
             splitFiles),
         BaseCombinedScanTask::new);
   }
+
+  public static Map<StructLikeWrapper, Collection<FileScanTask>> groupTasksByPartition(
```

Comment:
maybe I missed some other place, I only see it used in the strategy class, why is it not a private method?

Reply:
Moving the common `groupTasksByPartition` between [BaseRewriteDataFilesAction](https://github.com/apache/iceberg/blob/a1bd63d56751999ecee89b871992d7bac395fd52/core/src/main/java/org/apache/iceberg/actions/BaseRewriteDataFilesAction.java#L256) and [ConvertEqDeletesStrategy](https://github.com/apache/iceberg/pull/2364/files#diff-b31571d37b1152b422d32af62978f2954359254ac11a7f4338e3116b1b262c4cR137) into TableScanUtil should be OK for me.